### ISSUE TITLE
Naming/MemoizedInstanceVariableName is too aggressive for Rails controllers

### ISSUE THREAD

**Issue by hopsoft** [testdouble/standard#20]
I generally think this is a good practice; however, consider the following Rails controller action.

```ruby
class ExamplesController < ApplicationController
  before_action :set_example, except: [:index]
  
  ...

  def set_example
    @example ||= Example.find(params[:id])
  end
end
```

This will produce the following error.

```
Naming/MemoizedInstanceVariableName: Memoized variable @example does not match method name set_example. Use @set_example instead.
    @example ||= Example.find(params[:id])
    ^^^^^^^^
```

**Comment by JoshTGreenwood** [comment]
For this particular case, is there a reason `set_example` needs to be run before each action? 

In my experience, `before_action`s get written this way when the calling code interacts with both the method (`set_example`), and the instance variable `@example`. If all of the dependent code called the `example` method instead, it wouldn't necessarily need to be set in a before action or named like this.

I could very well be misinterpreting the critique though :)

**Comment by searls** [comment]
FWIW, I'm on board with this critique. I will look at it in the morning alongside some of the others we have so far 💚

**Comment by jasonkarns** [comment]
So far I agree with everything in this thread. It's such a common pattern in Rails controllers that it would be painful to use standard in Rails. But also, as @JoshTGreenwood mentioned, it's a pattern I try to refactor away from if I can. 🤷‍♂️

**Comment by hopsoft** [comment]
Agree that `before_actions` like this are something of an anti-pattern, but they are also a pretty common and idiomatic Railsism.

**Comment by searls** [comment]
Yeah, I don't think standard's goal is to peddle nuanced design critiques we will definitely make this change

**Comment by searls** [comment]
Landed in 0.0.11

**Comment by searls** [comment]
Landed in 0.0.11

**Comment by hopsoft** [comment]
Looks like this has resurfaced via a different rule.

```
Naming/AccessorMethodName: Do not prefix reader method names with get_.
```

**Comment by searls** [comment]
@hopsoft good point. Disabled that one too and landed in 0.1.6
